tests/pipeline/test_filter.py

```diff
@@ -379,6 +380,154 @@ def test_isfinite(self):
+                            [1, 1, 1, 1, 1, 1]], dtype=bool)
+
+        expected_4 = array([[0, 0, 0, 0, 1, 1],
+                            [1, 0, 0, 0, 0, 1],
+                            [1, 1, 0, 0, 0, 0],
+                            [1, 1, 1, 0, 0, 0],
+                            [1, 1, 1, 1, 0, 0],
+                            [1, 1, 1, 1, 1, 0],
+                            [1, 1, 1, 1, 1, 1]], dtype=bool)
+        self.check_terms(
+            terms={
+                '3': NoMissingValues([input_factor], window_length=3),
+                '4': NoMissingValues([input_factor], window_length=4),
+            },
+            expected={
+                '3': expected_3,
+                '4': expected_4,
+            },
+            initial_workspace={input_factor: data},
+            mask=self.build_mask(ones(shape=shape))
+        )
+
+    def test_no_missing_values_int_factor_input(self):
+        """Test int factor input to `NoMissingValues`
+        """
+        class SomeWindowSafeIntFactor(Factor):
+            dtype = int64_dtype
+            inputs = ()
+            window_length = 0
+            window_safe = True
+            missing_value = 0
+
+        input_factor = SomeWindowSafeIntFactor()
+
+        shape = (10, 6)
+        data = choice(range(1, 5), size=shape, replace=True)
+        data[eye(*shape, dtype=bool)] = input_factor.missing_value
+
+        expected_3 = array([[1, 0, 0, 0, 1, 1],
+                            [1, 1, 0, 0, 0, 1],
+                            [1, 1, 1, 0, 0, 0],
+                            [1, 1, 1, 1, 0, 0],
+                            [1, 1, 1, 1, 1, 0],
+                            [1, 1, 1, 1, 1, 1],
+                            [1, 1, 1, 1, 1, 1]], dtype=bool)
+
+        expected_4 = array([[0, 0, 0, 0, 1, 1],
+                            [1, 0, 0, 0, 0, 1],
+                            [1, 1, 0, 0, 0, 0],
+                            [1, 1, 1, 0, 0, 0],
+                            [1, 1, 1, 1, 0, 0],
+                            [1, 1, 1, 1, 1, 0],
+                            [1, 1, 1, 1, 1, 1]], dtype=bool)
+        self.check_terms(
+            terms={
+                '3': NoMissingValues([input_factor], window_length=3),
+                '4': NoMissingValues([input_factor], window_length=4),
+            },
+            expected={
+                '3': expected_3,
+                '4': expected_4,
+            },
+            initial_workspace={input_factor: data},
+            mask=self.build_mask(ones(shape=shape))
+        )
+
+    def test_no_missing_values_classifier_input(self):
+        """Test classifier factor input to `NoMissingValues`
+        """
+        class SomeWindowSafeStringClassifier(Classifier):
+            dtype = object_dtype
+            inputs = ()
+            window_length = 0
+            missing_value = ''
+            window_safe = True
+
+        input_factor = SomeWindowSafeStringClassifier()
+
+        shape = (10, 6)
+        data = choice(
```

Comment:
I think this should be a LabelArray if we're trying to simulate what the term would actually get passed?

Reply:
I think classifier inputs gets converted to a LabelArray when it is an input to `compute()`

Reply:
The place that data from loaders is normally converted into a LabelArray is in `zipline.lib.adjusted_array._normalize_array`, but that's not happening here because this test suite doesn't use any pipeline loaders, it just pre-populates the graph with the values provided by the test suite.

Reply:
Turns out this **is** happening, in `PipelineEngine._inputs_for_term`.